src/include/stir/IO/GEHDF5Wrapper.h

```diff
@@ -54,17 +56,33 @@ class GEHDF5Wrapper
 
     static bool check_GE_signature(const std::string& filename);
 
+    static GE_scanner check_current_signature(H5::H5File& file);
+
     explicit GEHDF5Wrapper();
 
     explicit GEHDF5Wrapper(const std::string& filename);
 
+    Succeeded check_file(); // AB todo, maybe private makes more sense?
+
+    
+
+    // bool is_list_file(const std::string& filename);
+
+    bool is_list_file();
+
+    bool is_sino_file();
+
+    bool is_geo_file();
+
+    bool is_norm_file();
+
     Succeeded open(const std::string& filename);
 
     Succeeded initialise_listmode_data(const std::string& path = "");
 
     Succeeded initialise_singles_data(const std::string& path = "");
 
-    Succeeded initialise_proj_data_data(const std::string& path = "",
+    Succeeded initialise_proj_data(const std::string& path = "",
```

Comment:
I believe this is now 1-based. Should be documented. No reason to have a default for the `view` (more dangerous than anything else)

Reply:
@KrisThielemans correct, but then path can not have default (only last parameters can have defaults). To be honest, all those paths are unused and I think unnecessary....

Reply:
if we don't need them, get rid of them. simpler is better.